### ISSUE TITLE
Fix broken publisher win notifications to show up again

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#1781]
## Bug Description

Our publisher win notifications have been broken for several releases at least. I hadn't been seeing them for months, and earlier finally took the time to figure out what's going on (a bunch of problems).

This issue is to address these bugs so that these notifications show up again like they used to. Note that we're not going to refactor anything here at this point, publisher wins will remain legacy for the foreseeable future. Improving our mechanisms for notifications overall (which will include this) is likely somewhere down the line in late 2020 or early 2021. We're not going to add any new publisher win notifications until then.

TLDR: There are still very fundamental architectural issues with this code, but this is not going to be addressed soon. We'll need to keep in mind to consider these notifications when making changes to things like our API calls, datapoints, API caching etc. which is clearly not DRY but necessary for now. The good thing after investing a few hours into this code is I finally have a grasp of how these things (somehwat) work at the moment 😆 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Publisher wins should show up on the dashboard again as expected:
    * On a fresh site with exactly 1 post, after activating Site Kit and authenticating you should see the notification "Congrats on your first post!".
    * On a fresh site with exactly 5 posts, after activating Site Kit and authenticating you should see the notification "Congrats on five published posts", accompanied by some Search Console stats.
    * On a site with more than 5 posts, after activating Site Kit and authenticating you should see the notification "Welcome Back!", accompanied by some Search Console stats.
    * With Analytics active, if your site's page views over the last month increased more than 10% compared to the month before, you should see the notification "Increased page views!", accompanied by some Analytics stats.
    * With Analytics active, if your site's unique visitors over the last month increased between 10% and 100% compared to the month before, you should see the notification "Congrats on more website visitors!", accompanied by some Analytics stats.
* All of these notifications should be dismissable and in this case they shouldn't show next time. (Note that only one of them will show at the same time.)

## Implementation Brief

* Fix duplicate JS action callback name `googlesitekit.dataLoadedGetNotifications` which was used by two different hooks. Both of these remove themselves after running once, however due to the duplicate name one would also remove the other. This prevented publisher win logic from running entirely.
* Update publisher win request definitions in `assets/js/components/publisher-wins/index.js` to match the updates made to the request definitions used for dashboard components, to be more precise add the `url` argument based on `_googlesitekitLegacyData.permaLink`. This is necessary because publisher wins always access data from cache, never via their own batch request. Since we made the `url` changes there a few releases ago, since then the publisher wins would never find the correct cache keys because the hashes became different.
* Update individual callbacks in `assets/js/components/publisher-wins/` files to expect the correct response format where the API data is in the root object (i.e. there is no `data.reports` or `data.rows` anymore). Due to this bug, even if the cache lookup had worked, we would not have seen publisher win notifications.
* Fix one formatting issue for the 'publishing-win' win notification: Since `extractSearchConsoleDashboardData` already uses `readableLargeNumber`, it doesn't need to be reused here. The duplicate usage actually resulted in some numbers being rendered as `0`, since after #1279 `readableLargeNumber` may return values that are not technically numeric (e.g. "12K").
* Merge #1782 (this was really mostly trial and error approach for fixing the various bugs, so by looking at the individual commits there things might become more clear).

## QA Brief

* 

## Changelog entry

* Fix publisher win notifications regression where they would not be displayed anymore.

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by adamsilverstein** [comment]
Nice work catching the issues here. In regards to https://github.com/google/site-kit-wp/issues/1540 when I reproduced this locally and looked at the data, I did have a notification that should have been showing but was not (so the [1] was right). So I believe fixing this issue will also fix that issue's core problem.

That ticket does raise the issue that users are confused by the red number which they associate with updates. It might be worth considering showing a red dot new new alerts. Also, I feel like _visiting the Site Kit dashboard_ should clear the notification - even if I don't dismiss the notification, the dot or number should go away as soon as I have seen it by visiting the dashboard.

**Comment by cole10up** [comment]
**Tested**

Installed latest SK develop, began running tests around posts. Noticed 5 post notification is showing an odd value for Average CTR below:

![image](https://user-images.githubusercontent.com/57115907/88260341-0548fd00-cc79-11ea-9b27-22c7d1247afb.png)

Sending back to adjust @felixarntz

**Comment by cole10up** [comment]
**Retested**

With a fresh Jurassic Ninja site, removed all posts:
![image](https://user-images.githubusercontent.com/57115907/88883313-e0a8d400-d1e8-11ea-8078-85ff42fd1da9.png)

Installed latest release candidate zip, activated and setup SK
![image](https://user-images.githubusercontent.com/57115907/88883547-7cd2db00-d1e9-11ea-8dff-4aa81dc9474e.png)

Posted a first post:
![image](https://user-images.githubusercontent.com/57115907/88883592-983de600-d1e9-11ea-93b3-2857d777830f.png)

Clicked ok got it
![image](https://user-images.githubusercontent.com/57115907/88883615-a5f36b80-d1e9-11ea-90fe-d068f618c7f7.png)

Refreshed and the banner hasn't come back

Posted 5 posts:
![image](https://user-images.githubusercontent.com/57115907/88883720-dcc98180-d1e9-11ea-8fc6-878e8e356881.png)

Confirmed 5 post message:
![image](https://user-images.githubusercontent.com/57115907/88883738-e7841680-d1e9-11ea-9729-b970a725d6a9.png)

Unable to accurately test the analytics updates in a timely manner, calling this one good based on the above tests.

Passed QA ✅
